lnbits/core/views/api.py

```diff
@@ -466,7 +466,7 @@ async def api_payment(payment_hash, X_Api_Key: Optional[str] = Header(None)):
     # We use X_Api_Key here because we want this call to work with and without keys
     # If a valid key is given, we also return the field "details", otherwise not
     wallet = await get_wallet_for_key(X_Api_Key) if isinstance(X_Api_Key, str) else None
-
+    wallet = wallet if wallet and not wallet.deleted else None
```

Comment:
shouldn't all these checks for the deleted flag be moved into the crud methods? Or is there any case where you would want to retrieve a deleted wlalet?